clj failing when archiving post with Emoji

I'm trying to download the lj "sodyera", and I get:

exec LJ.XMLRPC.getevents{:journal "sodyera", :selecttype "syncitems", :lastsync "2014-02-28 11:37:06", :get_video_ids true}
[Fatal Error] :1:3343: An invalid XML character (Unicode: 0xd83d) was found in the element content of the document.

Based on the posts that are downloaded before this one, I assume the post it's trying to download is https://sodyera.livejournal.com/461049.html, which contains Emoji.

It appears that this post is the oldest post in this LiveJournal with an Emoji (or any non-Base Multilingual Plane) character in it.  The first Emoji in the post is 💰, i.e. U+1f4b0, whose UTF-16 representation is 0xd83d 0xdcb0.

So my guess is that some part of the code is feeding UTF-16, and another part is expecting UTF-32, so it chokes on Unicode Surrogate Pairs.

The full stack is:

```
exec LJ.XMLRPC.getevents{:journal "sodyera", :selecttype "syncitems", :lastsync "2014-02-28 11:37:06", :get_video_ids true}
[Fatal Error] :1:3343: An invalid XML character (Unicode: 0xd83d) was found in the element content of the document.ERROR:  'An invalid XML character (Unicode: 0xd83d) was found in the element content of the document.'
org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 3343; An invalid XML character (Unicode: 0xd83d) was found in the element content of the document.
javax.xml.transform.TransformerException: org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 3343; An invalid XML character (Unicode: 0xd83d) was found in the element content of the document.
	at com.sun.org.apache.xalan.internal.xsltc.trax.TransformerImpl.transform(TransformerImpl.java:740)
	at com.sun.org.apache.xalan.internal.xsltc.trax.TransformerImpl.transform(TransformerImpl.java:343)
	at clojure.data.xml.jvm.pprint$indent_xml.invokeStatic(pprint.clj:27)
	at clojure.data.xml.jvm.pprint$indent_xml.invoke(pprint.clj:23)
	at clojure.data.xml$indent.invokeStatic(xml.clj:158)
	at clojure.data.xml$indent.doInvoke(xml.clj:153)
	at clojure.lang.RestFn.invoke(RestFn.java:425)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at clojure.core$apply.invokeStatic(core.clj:661)
	at clojure.core$apply.invoke(core.clj:652)
	at clojure.data.xml$indent_str.invokeStatic(xml.clj:164)
	at clojure.data.xml$indent_str.doInvoke(xml.clj:160)
	at clojure.lang.RestFn.invoke(RestFn.java:410)
	at clj.core$save_post.invokeStatic(core.clj:133)
	at clj.core$save_post.invoke(core.clj:126)
	at clj.core$run$fn__4792.invoke(core.clj:143)
	at clojure.lang.ArraySeq.reduce(ArraySeq.java:114)
	at clojure.core$reduce.invokeStatic(core.clj:6747)
	at clojure.core$reduce.invoke(core.clj:6730)
	at clj.core$fetch_all_posts.invokeStatic(core.clj:74)
	at clj.core$fetch_all_posts.invoke(core.clj:69)
	at clj.core$run.invokeStatic(core.clj:139)
	at clj.core$run.invoke(core.clj:136)
	at clj.core$_main.invokeStatic(core.clj:179)
	at clj.core$_main.doInvoke(core.clj:165)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clj.core.main(Unknown Source)
Caused by: org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 3343; An invalid XML character (Unicode: 0xd83d) was found in the element content of the document.
	at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1239)
	at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:643)
	at com.sun.org.apache.xalan.internal.xsltc.trax.TransformerImpl.transformIdentity(TransformerImpl.java:632)
	at com.sun.org.apache.xalan.internal.xsltc.trax.TransformerImpl.transform(TransformerImpl.java:728)
	... 27 more
---------
org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 3343; An invalid XML character (Unicode: 0xd83d) was found in the element content of the document.
	at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1239)
	at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:643)
	at com.sun.org.apache.xalan.internal.xsltc.trax.TransformerImpl.transformIdentity(TransformerImpl.java:632)
	at com.sun.org.apache.xalan.internal.xsltc.trax.TransformerImpl.transform(TransformerImpl.java:728)
	at com.sun.org.apache.xalan.internal.xsltc.trax.TransformerImpl.transform(TransformerImpl.java:343)
	at clojure.data.xml.jvm.pprint$indent_xml.invokeStatic(pprint.clj:27)
	at clojure.data.xml.jvm.pprint$indent_xml.invoke(pprint.clj:23)
	at clojure.data.xml$indent.invokeStatic(xml.clj:158)
	at clojure.data.xml$indent.doInvoke(xml.clj:153)
	at clojure.lang.RestFn.invoke(RestFn.java:425)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at clojure.core$apply.invokeStatic(core.clj:661)
	at clojure.core$apply.invoke(core.clj:652)
	at clojure.data.xml$indent_str.invokeStatic(xml.clj:164)
	at clojure.data.xml$indent_str.doInvoke(xml.clj:160)
	at clojure.lang.RestFn.invoke(RestFn.java:410)
	at clj.core$save_post.invokeStatic(core.clj:133)
	at clj.core$save_post.invoke(core.clj:126)
	at clj.core$run$fn__4792.invoke(core.clj:143)
	at clojure.lang.ArraySeq.reduce(ArraySeq.java:114)
	at clojure.core$reduce.invokeStatic(core.clj:6747)
	at clojure.core$reduce.invoke(core.clj:6730)
	at clj.core$fetch_all_posts.invokeStatic(core.clj:74)
	at clj.core$fetch_all_posts.invoke(core.clj:69)
	at clj.core$run.invokeStatic(core.clj:139)
	at clj.core$run.invoke(core.clj:136)
	at clj.core$_main.invokeStatic(core.clj:179)
	at clj.core$_main.doInvoke(core.clj:165)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clj.core.main(Unknown Source)
```

--- a/src/clj/xmlrpc.clj
+++ b/src/clj/xmlrpc.clj
@@ -8,7 +8,7 @@
   (:import org.apache.commons.codec.binary.Base64))
 
 (defn safe-xml [s]
-  (clojure.string/replace s #"[^\u0009\r\n\u0020-\uD7FF\uE000-\uFFFD\ud800\udc00-\udbff\udfff]" ""))
+  (clojure.string/replace s #"[^\u0009\r\n\u0020-\uFFFD\x{10000}-\x{10ffff}]" ""))
 
 (defmethod xml-rpc-value/parse-value :base64 [v]
   (safe-xml (String. (Base64/decodeBase64 ^String (text v)) "UTF-8")))

